# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieStorage.h
# Commit: 6379a4984eba
# Full Hash: 6379a4984ebafb659428ec3e4493f72173b3a70a
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-22 04:22:03
# Description:
#   Bug 1854028 - Update BounceTrackingStorageObserver cookie check and remove nsICookieNotification isThirdPartyCookie. r=bvandersloot,cookie-reviewers,anti-tracking-reviewers,timhuang
#   
#   For the BounceTrackingStorageObserver, instead of passing in a third-party flag
#   we can simply check if the changed cookie is partitioned or not.
#   
# ==============================================================================

diff -r 24c5f04d3793 -r 6379a4984eba netwerk/cookie/CookieStorage.h
--- a/netwerk/cookie/CookieStorage.h	Thu Sep 21 14:30:17 2023 +0000
+++ b/netwerk/cookie/CookieStorage.h	Thu Sep 21 14:42:32 2023 +0000
@@ -117,15 +117,13 @@
                      nsICookieNotification::Action aAction,
                      const nsACString& aBaseDomain,
                      dom::BrowsingContext* aBrowsingContext = nullptr,
-                     bool aIsThirdPartyCookie = false,
                      bool aOldCookieIsSession = false);
 
   void AddCookie(nsIConsoleReportCollector* aCRC, const nsACString& aBaseDomain,
                  const OriginAttributes& aOriginAttributes, Cookie* aCookie,
                  int64_t aCurrentTimeInUsec, nsIURI* aHostURI,
                  const nsACString& aCookieHeader, bool aFromHttp,
-                 dom::BrowsingContext* aBrowsingContext,
-                 bool aIsThirdPartyCookie);
+                 dom::BrowsingContext* aBrowsingContext);
 
   static void CreateOrUpdatePurgeList(nsCOMPtr<nsIArray>& aPurgedList,
                                       nsICookie* aCookie);